### PR TITLE
Create endpoint /api/leaderboard

### LINKED ISSUE #29: Add a web leaderboard
Similar to what we have with plusplus today, we should have a web version of the leaderboard where we can see points, and maybe some other stats (monthly, yearly, all time).

### PR DESCRIPTION
**Pull request by gueroverde**
Good morning team 

the approach on this PR was to split the application into 2 "kinds" of application, one to answer the request for the bot (BotApplication), and for response any URL that started with /api (ApliApplication). 

- Now when you use the URL `UrlOfApplication/api/leaderboard` you receive a JSON
- I added two variables to `.env` file `ENVIRONMENT` and `DEBUG` don't forget to add on your `.env`
- the next package was added: "symfony/http-kernel", "symfony/routing","symfony/framework-bundle": "5.1.8", don't forget run composer install before to start testing 
- created  `ContainerBuilderFactory` to allow used the creation of the container in wherever part of the application, and used on functional testing, allow us to use the container inside the testing folder.

At this moment I missing some tests that it's the reason this it's a draft, but please review and make any comments, any comment is welcome 😄

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -3,3 +3,5 @@ NGROK_HOSTNAME=
 NGROK_PORT=bot:8080
 SLACK_TOKEN=xoxb-...
 DB_FILE=db/phpmxbot.db
+ENVIRONMENT=prod
+DEBUG=false
```
Comment (jledezma-mt) [inline .env.example:7]: Is this used somewhere?

Comment (gueroverde) [reply .env.example:7]: Hey good morning,  it's using inside of the definition of the container https://github.com/phpmx/phpmxbot/pull/37/files#diff-c47b03734cd2b4337b345eeba955637ba790d51fd98979f6d366d4acbdb104e0R46

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -37,5 +37,8 @@ jobs:
         if: steps.composer-cache.outputs.cache-hit != 'true'
         run: composer install --prefer-dist --no-progress --no-suggest
 
+      - name: Copy .env.example
+        run: cp .env.example .env
+
```
Comment (eruizdechavez) [inline .github/workflows/php.yml:42]: Non-blocking, Question. Not sure why we need to generate a .env file. The application must default to use whatever environment variable is set with or without this file. This .env file is mostly used for development purposes.

File: .env.example (hunk 1 above)
Comment (eruizdechavez) [inline .env.example:7]: Non-blocking, Question. I do not know if this is a Symphony thing but, if this `DEBUG` variable is used to control the log output, the name  should be used LOG_LEVEL instead and the values usually are in the form of `info`, `notice`, `warning`, `error`, critical`, etc.

**Review by eruizdechavez — APPROVED** [review]
Just a couple of non-blocking questions. Feel free to merge at your earliest convenience.

File: .github/workflows/php.yml (hunk 2 above)
Comment (gueroverde) [reply .github/workflows/php.yml:42]: yes this file probably is not needed we can explore deleting and defining the environment variables (in the next PRS) on this file in other way, good catch

File: .env.example (hunk 1 above)
Comment (gueroverde) [reply .env.example:7]: I think this it's no for logging it's only how the application show the error pages, for example if this value is true only show the error and that it, in other case if this false, should detail that can help us to develop and find potentially bugs

**Pull request merged**